docs/eBPFWeakReferences.md

```diff
@@ -0,0 +1,82 @@
+3. A weak-ref object can be disassociated from the primary object it refers to.  This disassociation happens when the primary object is being destroyed, i.e. when the primary object's ref-count goes to zero. Note that the primary object's destruction does not affect the weak-ref's ref-count in any manner.
+4. Once dis-associated, a weak-ref object cannot be made to refer to it's previously referred to or another object.
+5. The life time of a weak-ref object is independent of the referred-to primary object and will always be greater than that of the referred-to object. To this end, the weak-ref object maintains it's own reference count, completely separate from that of the referred-to object.
+
+The ref-count manipulation for both, the weak-ref and the referred-to objects can be seen in the table below for the applicable events:
+
+```
+| Event                              | Weak-ref object   | Referred-to object      |
+|                                    | ref count         | ref count               |
+|------------------------------------|-------------------|-------------------------|
+| Referred-to Obj creation           | 1                 | 1                       |
+|   (weak-obj is created alongside)  |                   |                         |
+|                                    |                   |                         |
+| 'borrow' weak-ref, (get ptr to     | increment         | unchanged               |
+|   weak-ref associated with         |                   |                         |
+|   referred-to obj)                 |                   |                         |
+|                                    |                   |                         |
+| try-acquire referred-to            | unchanged         | increment, iff non-zero |
+|   obj via weak-ref                 |                   |                         |
+|                                    |                   |                         |
+| release referred-to obj ref        | unchanged         | decrement               |
+|   (decrements<br>ref count of      |                   |                         |
+|   referred-to obj)                 |                   |                         |
+|                                    |                   |                         |
+| weak-ref acquire ref               | increment         | unchanged               |
+|                                    |                   |                         |
+| weak-ref release ref               | decrement         | unchanged               |
+|                                    |                   |                         |
+| Referred-to obj<br>destruction     | decrement         | decrement               |
+|                                    |                   |                         |
+| Weak-ref obj<br>destruction        | decrement         | decrement               |
```

Comment:
Incorrect. Weak ref is by definition in the disassociated state when it is destroyed, so it can not possibly influence the referred to object.

Reply:
silly typo on my part. fixed